components/org.wso2.carbon.identity.api.server.claim.management/org.wso2.carbon.identity.rest.api.server.claim.management.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/claim/management/v1/core/ServerClaimManagementService.java

```diff
@@ -0,0 +1,796 @@
+        }
+
+    }
+
+    /**
+     * Update claim dialect.
+     * Due to the current implementation of the resource identifier of a new resource,
+     * the resource id of the claim dialect changes when an update occurs.
+     *
+     * @param dialectId          dialectId.
+     * @param claimDialectReqDTO claimDialectReqDTO.
+     * @return Resource identifier.
+     */
+    public String updateClaimDialect(String dialectId, ClaimDialectReqDTO claimDialectReqDTO) {
+
+        try {
+
+            // If the old and new dialect uri is the same we don't need to do a db update.
+            if (!StringUtils.equals(base64DecodeId(dialectId), claimDialectReqDTO.getDialectURI())) {
+                getClaimMetadataManagementService().renameClaimDialect(
+                        createClaimDialect(base64DecodeId(dialectId)),
+                        createClaimDialect(claimDialectReqDTO),
+                        ContextLoader.getTenantDomainFromContext());
+            } else {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Skipping db update as the old dialectURI and the new dialectURI is the same. " +
+                            "DialectURI: " + claimDialectReqDTO.getDialectURI());
+                }
+            }
+        } catch (ClaimMetadataException e) {
+            throw handleClaimManagementException(e, ERROR_CODE_ERROR_UPDATING_DIALECT);
+        }
+
+        // Since the dialects identifier has changed we have to send the new identifier in the location header.
+        return getResourceId(claimDialectReqDTO.getDialectURI());
+    }
+
+    /**
+     * Add a local claim.
+     *
+     * @param localClaimReqDTO localClaimReqDTO.
+     * @return Resource identifier.
+     */
+    public String addLocalClaim(LocalClaimReqDTO localClaimReqDTO) {
+
+        try {
+
+            for (AttributeMappingDTO attributeMappingDTO :
+                    localClaimReqDTO.getAttributeMapping()) {
+                if (!isUserStoreExists(attributeMappingDTO.getUserstore())) {
+                    throw handleClaimManagementClientError(ERROR_CODE_INVALID_USERSTORE, BAD_REQUEST);
+                }
+            }
+
+            getClaimMetadataManagementService().addLocalClaim(
+                    createLocalClaim(localClaimReqDTO),
+                    ContextLoader.getTenantDomainFromContext());
+        } catch (ClaimMetadataException e) {
+            throw handleClaimManagementException(e, ERROR_CODE_ERROR_ADDING_LOCAL_CLAIM);
+        } catch (UserStoreException e) {
+            throw handleException(e, ERROR_CODE_ERROR_ADDING_LOCAL_CLAIM);
+        }
+
+        return getResourceId(localClaimReqDTO.getClaimURI());
+    }
+
+    /**
+     * Delete a local claim.
+     *
+     * @param claimId claimId.
+     * @return True if delete was successful without any errors.
+     */
+    public boolean deleteLocalClaim(String claimId) {
```

Comment:
do we need boolean here?

Reply:
This is not necessary. Added to follow the pattern we have used when implementing the other server apis.